scripts/lib/libvirt/libvirt_setup.py

```diff
@@ -221,14 +229,26 @@ def net_start(args):
     netpath = args.netpath
     netname = xml_get_value(args.netpath, "name")
     print("defining network from {0}".format(netname))
-    networks = conn.listNetworks()
-    if netname not in networks:
-        print("defining network from {0}".format(netpath))
-        xml = readfile(netpath)
-        conn.networkDefineXML(xml)
-    print("starting {0} network".format(netname))
-    dom = conn.networkLookupByName(netname)
-    dom.create()
+
+    # Get the names of active and inactive network domains.
+    try:
```

Comment:
Looks like that all libvirt calls throw he same exception. IMHO catching it in the library is wrong.  A better approach is was proposed like 12 years ago by Guido:

https://www.artima.com/weblogs/viewpost.jsp?thread=4829

Basically: fail in the library, catch the exception in the `main()` block and use the `exit()` that you pointed in the `if` block:

```python
def main():
    try:
        ...
    except libvirt.libvirtError as e:
        # log / print something nice
        return 1
    except Except as e:
        # log / print something more
        return 2

if __name__ == '__main__':
    sys.exit(main())
```

This have two main advantages:

* The integration with the shell script is done in the `main()` level, that do not contain logic
* The `mkcloud` library can be reused, catching the exception in the client part and doing something sensible